websearch/websearch/src/durability.rs

```diff
@@ -192,11 +196,10 @@ mod durable_impl {
             }
         }
 
-        fn replay(original_params: SearchParams) -> Result<Self, SearchError> {
+        fn replay(replay_state: Impl::ReplayState) -> Result<Self, SearchError> {
             Ok(Self {
                 state: RefCell::new(Some(DurableSearchSessionState::Replay {
-                    original_params,
-                    current_page: 0,
+                    replay_state,
                     finished: false,
```

Comment:
you are currently not updating finished. It can be removed and you can rely on the underlying session (restored from the ReplayState) to handle finished correctly